empirical/scripts/calculate_empirical.py

```
@@ -0,0 +1,396 @@
+
+    Parameters
+    ----------
+    output_dir  : Path
+        Path to the output directory
+    ll_ffp  : Path
+        lat lon file path for the stations
+    vs30_ffp    : Path
+        vs30 file path for vs30 values at the stations
+    z_ffp   : Path
+        z file path for z1.0 and z2.5 values at the stations
+    srf_ffp : Path
+        srf file path for the fault data.
+    nhm_ffp : Path
+        nhm file path for the collection of faults data (default: NZ_FLTmodel_2010_v18p6.txt
+    srfdata_ffp : Path
+        srfdata file path for the fault data. Can be either relisation .csv or .info file
+    nz_gmdb_source_ffp  : Path
+        nz_gmdb_source file path for the source data
+    model_config_ffp    : Path
+        model_config file path for the empirical model. prescribes the model to be used for teconic class, IM and component
+    meta_config_ffp : Path
+        meta_config file path for the empirical model. prescribes the weight of the model for IM and Tectonic class
+    rjb_max : float
+        Maximum rupture distance
+    im_list : list
+        List of intensity measures. Currently supported: PGA, PGV, pSA, CAV, Ds575, Ds595
+    component   : str
+        Component of the IM (eg. geom, rotd50) to calculate empiricals for
+    periods : list
+        List of periods for pSA. Default is qcore.constants.DEFAULT_PSA_PERIODS
+    extended_period : bool
+        Indicate the use of extended(100) pSA periods
+
+    """
+    ### Data loading
+    model_config = utils.load_yaml(model_config_ffp)
+    # Using Full Loader for the meta config due to the python tuple pSA/PGA
+    meta_config = None
+    if meta_config_ffp is not None:
+        with open(meta_config_ffp) as f:
+            meta_config = yaml.load(f, Loader=yaml.FullLoader)
+
+    stations_df = formats.load_station_file(ll_ffp)
+    vs30_df = formats.load_vs30_file(vs30_ffp)
+    if z_ffp is not None and z_ffp.exists():
```

z_ffp is always set as it is a required field from the script. If the files doesn't exist surely an error is thrown instead?

---

> z_ffp is always set as it is a required field from the script. If the files doesn't exist surely an error is thrown instead?

z_ffp is not mandatory. If missing, we use a relation from vs30 to get estimated Z values. Old empirical engine code didn't accept the z_ffp and always used estimated values.

---

True, I misread and saw vs30 as required=True and thought it was the Z_ffp line, my bad

---

I think the z-file should be mandatory though, we pretty much always have Z-values these days, so I can't see a reason why we would every use a vs30->Z relation? Using these relations just has the potential for confusion imo.

Also this code always uses CY08 vs30->Z relation, there are many different of these relations depending both on the model and region. There is some existing code for this here
https://github.com/ucgmsim/Empirical_Engine/blob/master/empirical/util/z_model_calculations.py

---

Agreed that silent fallback to z-value estimation can cause potential confusion.
Will make the user specify `--z-estimate-with` from the list of available options. @claudio525